src/spi2parallel.cpp

```diff
@@ -0,0 +1,81 @@
+#include "spi2parallel.h"
+
+Spi2Parallel::Spi2Parallel(uint8_t pinCS, uint8_t deviceCount, uint8_t bitOrder)
+    : _pinCS(pinCS)
+    , _deviceCount(deviceCount)
+    , _bitOrder(bitOrder)
+    , _deviceChain(new uint8_t[_deviceCount - 1])
+    , _hardwareSPI(true) {
+}
+
+Spi2Parallel::Spi2Parallel(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t deviceCount, uint8_t bitOrder)
+    : _pinCS(pinCS)
+    , _pinMOSI(pinMOSI)
+    , _pinSCK(pinSCK)
+    , _deviceCount(deviceCount)
+    , _bitOrder(bitOrder)
+    , _deviceChain(new uint8_t[_deviceCount - 1])
+    , _hardwareSPI(false) {
+}
+
+Spi2Parallel::~Spi2Parallel() {
+    if (_deviceChain != nullptr) {
+        delete[] _deviceChain;
+    }
+}
+
+void Spi2Parallel::begin() {
+    pinMode(_pinCS, OUTPUT);
+    digitalWrite(_pinCS, LOW);
+    if (_hardwareSPI) {
+        //SPI.beginTransaction(SPISettings(1000000, _bitOrder, SPI_MODE0));
```

Comment:
safe it for future. SPI.setBitOrder now is deprecated, but SPI.beginTransaction is not working correctly

Reply:
Ok, this is described in the description:
https://www.arduino.cc/en/Reference/SPISetBitOrder